### PR TITLE
jenkins: scripted pipelines and shared funcs

### PR DESCRIPTION
**Pull request by habibayassin**

### CODE REVIEW

File: etc/DependencyInstaller.sh [hunk 1]
```diff
@@ -111,6 +111,39 @@ _installDarwinPackages() {
...
+    # apt-get -y install ca-certificates curl
+    # install -m 0755 -d /etc/apt/keyrings
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+
+    # curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+    # chmod a+r /etc/apt/keyrings/docker.asc
+    # echo \
+    # "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+    # $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
+    # tee /etc/apt/sources.list.d/docker.list > /dev/null
```
Comment (vvbandeira) [inline etc/DependencyInstaller.sh]:
```suggestion
    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
```

File: etc/DependencyInstaller.sh [hunk 2]
```diff
@@ -111,6 +111,39 @@ _installDarwinPackages() {
...
+    echo \
+    "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+    $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+    apt-get -y update
+    apt install -y docker-ce docker-ce-cli containerd.io
+    # apt-get -y install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```
Comment (vvbandeira) [inline etc/DependencyInstaller.sh]:
```suggestion
    apt-get install -y docker-ce docker-ce-cli containerd.io
```

File: etc/DependencyInstaller.sh [hunk 3]
```diff
@@ -111,6 +111,39 @@ _installDarwinPackages() {
...
+
+    # python
+    apt install -y python3
```
Comment (vvbandeira) [inline etc/DependencyInstaller.sh]:
```suggestion
    apt-get install -y python3
```

Comment (vvbandeira) [reply etc/DependencyInstaller.sh]: `apt-get` is prefered instead of `apt` as it works better with Docker.

File: etc/DependencyInstaller.sh [hunk 4]
```diff
@@ -220,6 +259,10 @@ case "${os}" in
         ;;
     "Ubuntu" )
         version=$(awk -F= '/^VERSION_ID/{print $2}' /etc/os-release | sed 's/"//g')
+        if [[ ${CI} == "yes" ]]; then
+            echo "Installing CI Tools"
+            _installCI
+        fi
```
Comment (vvbandeira) [inline etc/DependencyInstaller.sh:258]: Should have a message for unsupported OSes

File: etc/DockerHelper.sh [hunk 5]
```diff
@@ -79,34 +83,25 @@ _setup() {
...
-                done
-
-                for os in ${OS_LIST}; do
-                    echo [DRY-RUN] docker push openroad/flow-${os}-dev:latest
-                    echo [DRY-RUN] docker push openroad/flow-${os}-dev:${commitSha}
-                done
-
+            mkdir -p build
+            
+            docker login --username ${username} --password ${password}
```
Comment (vvbandeira) [inline etc/DockerHelper.sh]:
```suggestion
            mkdir -p build
            docker login --username ${username} --password ${password}
```

File: jenkins/public_tests_all_old.Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+      }
+    }
+
+    stage('Tests') {
+      matrix {
+        axes {
+          axis {
+            name 'TEST_SLUG';
+            values "docker build",
+                   "aes asap7",
```
Comment (vvbandeira) [inline jenkins/public_tests_all_old.Jenkinsfile]: Do you know if it is possible to read these values form one or multiple files?

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: jenkins/public_tests_all_old.Jenkinsfile (hunk 6 above)
Comment (habibayassin) [reply jenkins/public_tests_all_old.Jenkinsfile]: there's a possibility to define them as global variables configured in jenkins. There's another option to define a function in jenkins-ci that returns the list of test slugs.

**Pull request merged**